### PR TITLE
Use thiserror

### LINKED ISSUE #204: Errors: use https://lib.rs/crates/thiserror to reduce description duplication?
The doc updates sorta duplicate the meaning of the error message in the docstring _and_ the Display trait. Should we use thiserror in norad?

### PR DESCRIPTION
**Pull request by madig**
An experiment to see how using https://docs.rs/thiserror/latest/thiserror/ would look like.

Closes #204

### CODE REVIEW

**Comment by madig** [comment]
I suppose there are a few more errors in other modules that could be thiserror'ified if this is the approach we'd like.

File: src/write.rs [hunk 1]
```diff
@@ -166,16 +173,16 @@ mod tests {
         let opt = WriteOptions::default();
         let plist_read = Value::from_file("testdata/MutatorSansLightWide.ufo/lib.plist")
             .expect("failed to read plist");
-        let tmp = TempDir::new("test")?;
+        let tmp = TempDir::new("test").unwrap();
```
Comment (cmyr) [inline src/write.rs:176]: what's the motivation for these unwraps? If we're going to move to unwrapping then the test doesn't need to return a `Result`, but it seems easier to leave it as is?

**Review by cmyr — COMMENTED** [review]
I'm happy to see this merged, I took a few quick measurements and compile time does not seem to be significantly impacted.

File: src/write.rs (hunk 1 above)
Comment (madig) [reply src/write.rs:176]: I didn't want to add a new `From` converter thing, but haven't looked at it too closely :grimacing:

Comment (cmyr) [reply src/write.rs:176]: Sure, that's reasonable. Another option might be to make the return type `Box<dyn std::error::Error>`, but whatever is easiest. :)

**Review by cmyr — APPROVED** [review]
thanks!

**Pull request merged**